### PR TITLE
`geom_function()`: x-axis title produced automatically where no data is added

### LINKED ISSUE #5647: `geom_function` does not produce a x axis title where no data is added
``` r
library(tidyverse)

p <- ggplot() +
  geom_function(
    fun = \(x) dnorm(x),
  )

#currently produces this
p
```

``` r

#I think it should produce this
p +
  labs(x = "x")
```

<sup>Created on 2024-01-16 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

### PR DESCRIPTION
**Pull request by phispu**
Fixes #5647

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # ggplot2 (development version)
 
+* Fixed bug in `geom_smooth()` so x-axis title now produced automatically
```
Comment (thomasp85) [inline NEWS.md:3]:
```suggestion
* Fixed bug in `stat_function()` so x-axis title now produced automatically   
```

**Review by thomasp85 — CHANGES_REQUESTED** [review]
Thanks. Just a small change requested

**Comment by phispu** [comment]
Nice catch - updated!

**Review by thomasp85 — APPROVED** [review]
LGTM

**Pull request merged**